pilot/pkg/bootstrap/server.go

```diff
@@ -238,13 +244,15 @@ func NewServer(args *PilotArgs) (*Server, error) {
 		return nil, fmt.Errorf("error initializing secure gRPC Listener: %v", err)
 	}
 
-	// common https server for webhooks (e.g. injection, validation)
-	if err := s.initHTTPSWebhookServer(args); err != nil {
-		// Not crashing istiod - existing pods will keep working, new pods
-		// may fail if all webhook injectors are down.
-		// This typically happens if certs are missing.
-		log.Errorf("error initializing injection webhook server: %v", err)
+	// setup watches for certs - This has to be called after initSecureGrpcListener because it sets up DNS certs.
+	if err := s.initCertificateWatches(args.TLSOptions); err != nil {
+		// Not crashing istiod - This typically happens if certs are missing.
+		log.Errorf("error initializing certificate watches: %v", err)
```

Comment:
I think this is not correct now, as the secure grpc server need this certificate

Reply:
Ok. Let me convert this to a proper error